### PR TITLE
fix: path is null or empty error

### PR DESCRIPTION
**Pull request by cslong**
*Description of changes:*
Add defensive check for empty file path returned in solution analysis results. Also add retry so single exception won't block other diagnostics from returning.

*Testing done:*
Add unit test to check that empty file path won't throw exception. Existing testing passes

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/porting-assistant-dotnet-ui/blob/develop/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/porting-assistant-dotnet-ui/blob/develop/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including READMEs and comments (where appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific environment

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/PortingAssistantExtensionServer/Services/AnalysisService.cs [hunk 1]
```diff
@@ -193,13 +193,32 @@ public bool HasSolutionAnalysisResult()
                         });
                     }
 
+                    int numberOfExceptions = 0;
                     foreach (var sourceFileAnalysisResult in projectAnalysisResult.SourceFileAnalysisResults)
                     {
-                        var sourceFileUri = DocumentUri.FromFileSystemPath(sourceFileAnalysisResult.SourceFilePath);
-                        var diagnostics = GetDiagnostics(sourceFileAnalysisResult);
-                        if (!FileToFirstDiagnostics.ContainsKey(sourceFileUri))
+                        const int maxNumberOfExceptions = 5;
```
Comment (kanikaul-amazon) [inline src/PortingAssistantExtensionServer/Services/AnalysisService.cs:199]: nit: Can be stored in constants file

**Review by kanikaul-amazon — DISMISSED** [review]
Approved, with non blocking comment

**Review by huawenm — APPROVED** [review]

**Review by kanikaul-amazon — APPROVED** [review]

**Review by huawenm — APPROVED** [review]
